packages/db/src/lib/file.ts

```diff
@@ -0,0 +1,49 @@
+import Knex from 'Knex';
+
+import { tableNames } from '../tables';
+import { PostgresFileRecord } from '../types/file';
+import { PostgresGranuleRecord } from '../types/granule';
+
+/**
+ * Helper to build a query that returns records from the files table
+ * with data joined in from the granules table.
+ *
+ * @param {Object} params
+ * @param {Knex} params.knex - Knex client object
+ * @param {Partial<PostgresFileRecord>} params.searchParams
+ *   Query search parameters for files table
+ * @param {Array<string>} params.sortColumns
+ *   Columns to sort results by
+ * @param {Array<string>} params.granuleColumns
+ *   Columns to return from granules table
+ * @param {number} [params.limit] - Limit on number of results to return. Optional.
```

Comment:
```suggestion
 * @param {number} params.limit - Limit on number of results to return. Optional.
```
Does the `[]` signify that it's optional? Not sure if it's necessary

Reply:
Or perhaps all other params need them too?